x-pack/plugins/infra/public/components/asset_details/hooks/use_date_range.ts

```diff
@@ -5,33 +5,38 @@
  * 2.0.
  */
 
+import type { TimeRange } from '@kbn/es-query';
 import createContainer from 'constate';
-import { useMemo } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { parseDateRange } from '../../../utils/datemath';
-import type { AssetDetailsProps } from '../types';
+
 import { toTimestampRange } from '../utils';
 
-const DEFAULT_DATE_RANGE = {
+const DEFAULT_DATE_RANGE: TimeRange = {
   from: 'now-15m',
   to: 'now',
 };
 
-export type UseAssetDetailsStateProps = Pick<AssetDetailsProps, 'dateRange'>;
+export interface UseAssetDetailsStateProps {
+  initialDateRange: TimeRange;
+}
+
+export function useDateRangeProvider({ initialDateRange }: UseAssetDetailsStateProps) {
```

Comment:
When we have the date range persisted in the URL (the case of the flyout for example) should we first get the date range from the URL value (passed to the asset details component) and if we don't have it to fallback to default (the value selected in the host view / the 15m default)?

Reply:
You're completely right. I totally forgot about storing the date range in the URL state.